pkg/provisioner/ironic/raid.go

```diff
@@ -0,0 +1,173 @@
+package ironic
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/nodes"
+
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/devicehints"
+
+	"github.com/pkg/errors"
+)
+
+// setTargetRAIDCfg set the RAID settings to the ironic Node for RAID configuration steps
+func setTargetRAIDCfg(p *ironicProvisioner, ironicNode *nodes.Node) (err error) {
+	var logicalDisks []nodes.LogicalDisk
+
+	// Build target for RAID configuration steps
+	logicalDisks, err = buildTargetRAIDCfg(p.host.Status.Provisioning.RAID)
+	if len(logicalDisks) == 0 || err != nil {
+		return
+	}
+
+	// set root volume
+	if p.host.Status.Provisioning.RootDeviceHints == nil {
+		logicalDisks[0].IsRootVolume = new(bool)
+		*logicalDisks[0].IsRootVolume = true
+	} else {
+		p.log.Info("rootDeviceHints is used, the first volume of raid will not be set to root")
+	}
+
+	// Set target for RAID configuration steps
+	return nodes.SetRAIDConfig(
+		p.client,
+		ironicNode.UUID,
+		nodes.RAIDConfigOpts{LogicalDisks: logicalDisks},
+	).ExtractErr()
+}
+
+// A private method to build RAID disks
+func buildTargetRAIDCfg(raid *metal3v1alpha1.RAIDConfig) (logicalDisks []nodes.LogicalDisk, err error) {
```

Comment:
nit: maybe make this public for reuse? e.g. we use these bits in openshift installer (if you care about it).